libraries/chain/database.cpp

```diff
@@ -284,14 +285,18 @@ namespace golos { namespace chain {
                         }
 
                         auto end = fc::time_point::now();
+                        auto cur_block_pos = _block_log.get_block_pos(cur_block_num);
                         auto cur_block = *_block_log.read_block_by_num(cur_block_num);
 
-                        if (cur_block_num % 100000 == 0) {
+                        auto reindex_percent = cur_block_pos * 100 / last_block_pos;
+                        if (reindex_percent - last_reindex_percent >= 1) {
                             std::cerr
-                                << "   " << double(cur_block_num * 100) / last_block_num << "%   "
+                                << "   " << reindex_percent << "%   "
```

Comment:
For  checking percents it is good to use integer numbers.
But for information it is real number, it more informal. 

Return back the formula about % in output:
```<< "   " << double(cur_block_num * 100) / last_block_num << "%   "```

Reply:
But integer percents could simplify automated log parsing (real-time) if need...

Reply:
Ok. Your explanation looks reasonable.